.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Lint Go Code
+
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.19.0
+
+    - name: Install golangci-lint
+      run: go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+
+    - name: Run golangci-lint
+      run: golangci-lint run --allow-serial-runners
```

Comment:
I suggest moving this to `Makefile`, so the user can execute it locally to get lint output before submitting a PR and running the CI